xonsh/prompt/vc.py

```diff
@@ -52,23 +52,33 @@ def get_git_branch():
         branch = None
     return branch
 
+def _get_hg_root(q):
+    _curpwd = os.getcwd()
+    while True:
+        if any([b.name == '.hg' for b in os.scandir(_curpwd)]):
+            q.put(_curpwd)
+            break
+        else:
+            _oldpwd = _curpwd
+            _curpwd = os.path.split(_curpwd)[0]
+            if _oldpwd == _curpwd:
+                return False
```

Comment:
Yes. But that wasn't why it was changed. At least that is what I remember. The difference here is that @gforsyth runs it in a thread. Or am I misreading it?